src/app/functions/server/onerep.ts

```diff
@@ -117,12 +117,42 @@ async function onerepFetch(
   if (!onerepApiKey) {
     throw new Error("ONEREP_API_KEY env var not set");
   }
-  const url = new URL(path, onerepApiBase);
-  const headers = new Headers(options.headers);
-  headers.set("Authorization", `Bearer ${onerepApiKey}`);
-  headers.set("Accept", "application/json");
-  headers.set("Content-Type", "application/json");
-  return fetch(url, { ...options, headers });
+  try {
+    const url = `${onerepApiBase}${path}`;
+    const headers = new Headers(options.headers);
+    headers.set("Authorization", `Bearer ${onerepApiKey}`);
+    headers.set("Accept", "application/json");
+    headers.set("Content-Type", "application/json");
+
+    const response = await fetch(url, { ...options, headers });
+
+    if (!response.ok) {
+      const status = {
+        status: response.status,
+        statusText: response.statusText,
```

Comment:
Maybe it would be a good idea to include the URL here as well? Otherwise it's a bit hard to determine _which_ OneRep fetch failed.

```suggestion
        statusText: response.statusText,
        url: url,
```